src/login/logind.c

```diff
@@ -349,116 +350,88 @@ static int manager_enumerate_users(Manager *m) {
         return r;
 }
 
-static int parse_fdname(const char *fdname, char **session_id, dev_t *dev) {
+static int parse_fdname(const char *fdname, char **ret_session_id, dev_t *ret_devno) {
         _cleanup_strv_free_ char **parts = NULL;
         _cleanup_free_ char *id = NULL;
-        unsigned major, minor;
         int r;
 
+        assert(ret_session_id);
+        assert(ret_devno);
+
         parts = strv_split(fdname, "-");
         if (!parts)
                 return -ENOMEM;
-        if (strv_length(parts) != 5)
-                return -EINVAL;
 
-        if (!streq(parts[0], "session"))
-                return -EINVAL;
+        assert(streq(parts[0], "session"));
```

Comment:
I do not think we should use assert() for parsing string passed from outside. Of course, PID1 passed the string that generated by logind as is. But still, we should return negative errno in such case, and handle invalid string gracefully.